### PR TITLE
🐛 Make env take precedence over dotenv

### PR DESCRIPTION
**Pull request by muchzill4**
### Description

As per title.

https://pypi.org/project/python-dotenv/
> load_dotenv does not override existing System environment variables. To override, pass override=True to load_dotenv().

### Test Coverage

- [ ]  This change is covered by existing or additional automated tests.
- [x]  Manual testing has been performed (and evidence provided) as automated testing was not feasible.
- [ ]  Additional tests are not required for this change (e.g. documentation update).

### CODE REVIEW

File: mbed_targets/config.py [hunk 1]
```diff
@@ -17,7 +17,7 @@
 import os
 import dotenv
 
-dotenv.load_dotenv(dotenv.find_dotenv(usecwd=True), override=True)
+dotenv.load_dotenv(dotenv.find_dotenv(usecwd=True))
```
Comment (rwalton-arm) [inline mbed_targets/config.py]: Don't we need to actually read from the env at the call site? We cache the variable at import time so we could get out of sync with the user's shell environment?

Comment (muchzill4) [reply mbed_targets/config.py]: Isn't this true of this and previous approach where variables were assigned at load time? This has not been introduced by this PR, right?

Comment (rwalton-arm) [reply mbed_targets/config.py]: It wasn't true in the original code (i.e when this lived in `target_database`) as we read from the env in `_get_request`. Must've changed when it was moved here, and I missed it in the review.

Comment (muchzill4) [reply mbed_targets/config.py]: Could we address that separately then? I'd be keen not to pollute this PR with unrelated bugfix. :)

Comment (rwalton-arm) [reply mbed_targets/config.py]: Yeah, makes sense.

**Pull request merged**